### PR TITLE
Changed bin edges to points

### LINKED ISSUE #23792: Bin edges to points in RemoveExpDecay and EstimateMuonAsymmetryFromCounts
RemoveExpDecay and EstimateMuonAsymmetryFromCounts should both use point data instead of bin edges when removing the exponential. This is low priority as the bin widths are of the order ns so it makes little difference.

### PR DESCRIPTION
**Pull request by SilkeSchomann**
Fixes [#23792](https://github.com/mantidproject/mantid/issues/23792)

When removing the exponential in RemoveExpDecay and EstimateMuonAsymmetryFromCounts use point data instead of bin edges for the x values.

To Test: Run the algorithms EstimateMuonAsymmetryFromCounts and RemoveExpDecay.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DannyHindson** [comment]
@SilkeSchomann - I've just picked up this PR to review. We normally put a "**To test:**" section in the PR description. Would you mind adding this in? In some cases there genuinely isn't much to test in addition to the automated tests so in that case I normally just write "code review only" or similar. I think in this case it's probably just a case of running one of the two algorithms you mention in the description

File: Framework/Muon/src/RemoveExpDecay.cpp [hunk 1]
```diff
@@ -148,11 +148,12 @@ void MuonRemoveExpDecay::exec() {
 HistogramData::Histogram MuonRemoveExpDecay::removeDecay(
     const HistogramData::Histogram &histogram) const {
   HistogramData::Histogram result(histogram);
+  const auto xPointRawData = histogram.points().rawData();
```
Comment (DannyHindson) [inline Framework/Muon/src/RemoveExpDecay.cpp:151]: Sorry these two points are a bit pedantic...
- Please could you change this line so it reads the x points off the result object. The code used to read the bin edges from result and the Y and E values are also read off result so seems more consistent
- Also the Points class returned by points() already supports indexing using [i] so no need for the .rawData() bit on the end. Probably makes sense to rename the variable to xPoints or similar as part of this

**Review by DannyHindson — CHANGES_REQUESTED** [review]
I noticed that the unit test for RemoveExpDecay didn't need a change (RemoveExpDecayTest.h) despite the underlying algorithm changing. I had a closer look at why and the unit test always creates workspaces that use points rather than bin edges. I think it would be useful to add an extra test that runs on a workspace with bin edges. I think you can do that by adding a new boolean parameter called usePoints or similar to the createWorkspace function in RemoveExpDecayTest.h. If you create the X vector with one more entry in it than the Y vector the call to the CreateWorkspace algorithm will create a workspace using bin edges.

I've also a couple of detail level comments against the individual units.

File: Framework/Muon/src/MuonAsymmetryHelper.cpp [hunk 2]
```diff
@@ -51,10 +51,12 @@ HistogramData::Histogram
 normaliseCounts(const HistogramData::Histogram &histogram,
                 const double numGoodFrames) {
   HistogramData::Histogram result(histogram);
+  const auto xPointRawData = histogram.points().rawData();
```
Comment (DannyHindson) [inline Framework/Muon/src/MuonAsymmetryHelper.cpp:54]: See comments against RemoveExpDecay.cpp - please can you make same changes here

**Review by DannyHindson — CHANGES_REQUESTED** [review]

File: Framework/Muon/test/RemoveExpDecayTest.h [hunk 3]
```diff
@@ -32,6 +33,10 @@ MatrixWorkspace_sptr createWorkspace(size_t nspec, size_t maxt) {
   MantidVec X;
   MantidVec Y;
   MantidVec E;
+  if (useBinEdges) {
+    X.emplace_back(0.0);
+    X.emplace_back(0.0);
```
Comment (DannyHindson) [inline Framework/Muon/test/RemoveExpDecayTest.h:38]: Thanks for making the changes. All looks good apart from one little thing related to these extra x values. 
I think this code is currently adding both of the extra x values on the front of the first histogram. Also it assumes nspec is always equal to 2 ie always two histograms. I think you should move a single X.emplace_back(0.0) statement down to in between lines 40 and 41 and then that will work better. You might have to adjust the test values lower down as well where you check the x values at index 10, 19 etc are what you expect.

**Review by DannyHindson — CHANGES_REQUESTED** [review]

**Review by DannyHindson — CHANGES_REQUESTED** [review]
This looks good now. Please can you add a line into the Muon.rst release note in docs\source\release?

**Comment by SilkeSchomann** [comment]
The release note is already updated (https://github.com/mantidproject/mantid/pull/30101)

**Comment by DannyHindson** [comment]
> The release note is already updated (#30101)

OK. I don't think there's much benefit in separating the release note change into a separate pull request. Please can you make the change under this PR and close #30101? I think Jenkins will spot it is a docs change only and won't run all the jobs. I see the RHEL7 job has failed because of a problem with one of the agents so will restart that now

**Review by DannyHindson — APPROVED** [review]
Thanks for making the changes. Looks good to me

**Review by DavidFair — APPROVED** [review]
I had to double check that we weren't making an expensive copy with `const auto xPoints = result.points()` but looking at the implementation someone has already wrapped it in a light-weight container, so we don't need to worry about it which is a nice bonus

LGTM lets :shipit:

**Pull request merged**